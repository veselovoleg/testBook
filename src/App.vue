<template>
  <div id="app">
    <h1>Hello</h1>
    <MessageList :messages="messages"/>
  </div>
</template>

<script>
import MessageList from './components/MessageList'

export default {
  name: 'app',
  data: () => ({ messages: ['Hey John', 'Howdy Paco'] }),
  components: {
    MessageList
  }
}
</script>

<style>
#app {
  text-align: center;
}
</style>

