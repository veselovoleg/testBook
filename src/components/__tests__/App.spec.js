import { shallowMount } from '@vue/test-utils'
import App from '@/App.vue'

describe("App.test.js", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(App, {
            // Create a shallow instance of the component
            data: () => ({ messages: ['Cat'] }),
        });
    });

    it('equals messages to ["Cat"]', () => {
        // Within cmp.vm, we can access all Vue instance methods
        expect(wrapper.vm.messages).toEqual(["Cat"]);
    });
    
    it("has the expected html structure", () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});