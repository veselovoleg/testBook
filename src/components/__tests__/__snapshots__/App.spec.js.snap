// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <h1>
    Hello
  </h1>
   
  <messagelist-stub
    messages="Cat"
  />
</div>
`;
